frappe/utils/backups.py

```diff
@@ -104,14 +104,43 @@ def zip_files(self):
 
 			print('Backed up files', os.path.abspath(backup_path))
 
-	def take_dump(self):
+	def take_dump(self, encrypt=False, automated_backup=None):
 		import frappe.utils
 
 		# escape reserved characters
 		args = dict([item[0], frappe.utils.esc(item[1], '$ ')]
 			for item in self.__dict__.copy().items())
 
-		cmd_string = """mysqldump --single-transaction --quick --lock-tables=false -u %(user)s -p%(password)s %(db_name)s -h %(db_host)s > %(backup_path_db)s """ % args
+		cmd_string = ""
+		if not automated_backup:
+			if encrypt:
+				file_name = args['backup_path_db'].split('/')
+				file_name = file_name[4]
+				file_name = file_name.replace('.sql', '.xb.enc')
+				args['backup_path_db'] = args['backup_path_db'].replace(args['backup_path_db'].split('/')[4], file_name)
+				args['encryption_password'] = frappe.utils.password.get_decrypted_password("System Settings",
+																						   "System Settings",
+																						   "confirm_password_encryption",
+																						   False)
+				cmd_string = """mysqldump --single-transaction --quick --lock-tables=false -u %(user)s -p%(password)s %(db_name)s -h %(db_host)s --triggers --routines | openssl enc -aes-256-cbc -salt -k %(encryption_password)s > %(backup_path_db)s""" % args
+			else:
+				cmd_string = """mysqldump --single-transaction --quick --lock-tables=false -u %(user)s -p%(password)s %(db_name)s -h %(db_host)s > %(backup_path_db)s """ % args
+
+		elif hasattr(frappe.get_doc("System Settings"), "enable_backup_encryption"):
+			if frappe.get_doc("System Settings").enable_backup_encryption:
+				file_name = args['backup_path_db'].split('/')
+				file_name = file_name[4]
+				file_name = file_name.replace('.sql', '.xb.enc')
+				args['backup_path_db'] = args['backup_path_db'].replace(args['backup_path_db'].split('/')[4], file_name)
+				args['encryption_password'] = frappe.utils.password.get_decrypted_password("System Settings",
+																						   "System Settings",
+																						   "confirm_password_encryption",
+																						   False)
+
+				cmd_string = """mysqldump --single-transaction --quick --lock-tables=false -u %(user)s -p%(password)s %(db_name)s -h %(db_host)s --triggers --routines | openssl enc -aes-256-cbc -salt -k %(encryption_password)s > %(backup_path_db)s""" % args
```

Comment:
This looks similar to [this line](https://github.com/frappe/frappe/pull/7880/files#diff-0d4ce8a951300571b56f0af108724c06R125). Let's not have redundant code.

Reply:
this line is for the backup process coming from a cron job. the first condition that has this line ```cmd_string = """mysqldump --single-transaction --quick --lock-tables=false -u %(user)s -p%(password)s %(db_name)s -h %(db_host)s --triggers --routines | openssl enc -aes-256-cbc -salt -k %(encryption_password)s > %(backup_path_db)s""" % args``` is for the backup process done manually, the other the same line is for the cron job process.

Reply:
You're right, we should remove this line.

Reply:
actually, it's better to put this line on the second condition. 
```		elif automated_backup:
			if frappe.get_doc("System Settings").enable_backup_encryption:
				file_name = args['backup_path_db'].split('/')
				file_name = file_name[4]
				file_name = file_name.replace('.sql', '.xb.enc')
				args['backup_path_db'] = args['backup_path_db'].replace(args['backup_path_db'].split('/')[4], file_name)
				args['encryption_password'] = frappe.utils.password.get_decrypted_password("System Settings",
																						   "System Settings",
																						   "password_encryption",
																						   False)

				cmd_string = """mysqldump --single-transaction --quick --lock-tables=false -u %(user)s -p%(password)s %(db_name)s -h %(db_host)s --triggers --routines | openssl enc -aes-256-cbc -salt -k %(encryption_password)s > %(backup_path_db)s""" % args

			else:
				cmd_string = """mysqldump --single-transaction --quick --lock-tables=false -u %(user)s -p%(password)s %(db_name)s -h %(db_host)s > %(backup_path_db)s """ % args
```

Reply:
so if ever backup process is done through cron job, it will execute the condition for the automated backup and check if the encryption option is selected and if not that redundant line before will be used.